metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -327,13 +327,25 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 		return nil, err
 	}
 
-	cluster_name := ""
+	clusterName := ""
 	if len(opts["cluster_name"]) >= 1 {
-		cluster_name = opts["cluster_name"][0]
+		clusterName = opts["cluster_name"][0]
 	} else {
 		glog.Warning("Cluster name required but not provided, using empty cluster name.")
 	}
 
+	if clusterName == "" {
+		glog.Info("An empty cluster name has been provided, checking the GCE Metadata Server to try to auto-detect.")
+
+		clusterName_, err := gce.InstanceAttributeValue("cluster-name")
+		if err == nil {
+			glog.Infof("Discovered '%s' as the cluster name from the GCE Metadata Server.", clusterName_)
+			clusterName = clusterName_
+		} else {
+			glog.Warning("Cluster name could not be discovered using the GCE Metadata Server.")
```

Comment:
I would include error message in this warning.

Reply:
Updated